### DISCUSSION TITLE
Deploy Asciinema-server without namshi container

### DISCUSSION THREAD

**Discussion by mohatb** [asciinema/discussions#111] [Q&A] [answered]
Hi, is it possible to change the server to send emails without using namshi container?  for example SMTP AUTH client submission in office 365 ?

**Comment by colinleefish** [comment]
We managed to do so by changing the last part of `/config/prod.exs` and build a new docker image out of it.

The original config file looks like this.

```elixir
config :asciinema, Asciinema.Emails.Mailer,
  deliver_later_strategy: Asciinema.BambooExqStrategy,
  adapter: Bamboo.SMTPAdapter,
  server: "smtp",
  port: 25
```

And here's our modified config. Note that we are using Office 365 China version and you may change the address.

```elixir
config :asciinema, Asciinema.Emails.Mailer,
  deliver_later_strategy: Asciinema.BambooExqStrategy,
  adapter: Bamboo.SMTPAdapter,
  server: "smtp.partner.outlook.cn",
  port: 587,
  username: "a@b.com",
  password: "my-password",
  ssl: false,
  tls: :always,
  auth: :always,
  allowed_tls_versions: [:"tlsv1.2"],
  retries: 2
```

Note: To make it work, `ssl` needs to be "false" while `tls` needs to be ":always".

Further reading:
- https://github.com/fewlinesco/bamboo_smtp

**Comment by ku1ik** [comment]
I've created a wiki page about this, showing how this can be configured _without rebuilding the image_ - https://github.com/asciinema/asciinema-server/wiki/SMTP-configuration

Thx for the Office 365 snippet @colinleefish !

Does it solve your problem @mohatb , you ok with closing the issue?

**Comment by colinleefish** [comment]
> I've created a wiki page about this, showing how this can be configured _without rebuilding the image_ - https://github.com/asciinema/asciinema-server/wiki/SMTP-configuration
> 
> Thx for the Office 365 snippet @colinleefish !
> 
> Does it solve your problem @mohatb , you ok with closing the issue?

Glad that I can help!

**Comment by ku1ik** [comment, marked as answer]
You can now configure SMTP server directly on the asciinema-server with `SMTP_*` variables as documented here: https://docs.asciinema.org/manual/server/self-hosting/configuration/#email

No need for namshi/smtp containers, in fact I've removed it from the docs.
